sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -260,8 +267,10 @@ class BulkWriter(container: CosmosAsyncContainer,
 
   private def shouldIgnore(cosmosException: CosmosException): Boolean = {
     // ignore 409 on create-item
-    writeConfig.itemWriteStrategy == ItemWriteStrategy.ItemAppend &&
-      Exceptions.isResourceExistsException(cosmosException)
+    (writeConfig.itemWriteStrategy == ItemWriteStrategy.ItemAppend &&
+      Exceptions.isResourceExistsException(cosmosException)) ||
+      (writeConfig.itemWriteStrategy == ItemWriteStrategy.ItemDelete &&
```

Comment:
Would the `ItemDeleteIfNotModified` return 412 if the Etag passed is not the last one? Should we ignore in that case?

Would it be a bit more clear to have switch statements instead of ORs as the conditions keep increasing?

Reply:
Good catch!

Reply:
Fixed